Closure_Front_End/src/App.vue

```diff
@@ -42,9 +42,35 @@
 </template>
 
 <script>
+import axios from "axios";
+
 export default {
   name: "App",
 
+  data() {
+    return {
+      username: "",
+      student: null,
+    };
+  },
+
+  created() {
+    /** if the user is loged in, then fetching his data from DB, else doing nothing */
+    if (this.$auth.isAuthenticated.value) {
+      this.$auth
+        .getIdTokenClaims()
+        .then((response) => (this.username = response.nickname));
+
+      axios
+        .get("http://127.0.0.1:8000/api/v1/students/" + this.username, {
```

Comment:
In my last PR, I implemented the global propert `this.$http` which already includes an authorization header when you're logged in, as well as the base directory, so you can do the following instead:

```javascript
this.$http.get(`/students/${this.username}`)
```

Furthermore, since you want to run this function only after you're authenticated, which is not necssarily true at this point in the code(the above call is asynchronous), you might want to put this functionality somewhere else. For example, you can use a `watch`  on `this.$auth.isAuthenticated` to run a piece of code whenever we authenticate (e.g, if we loaded the page and we were already authenticated, or if we manually pressed the login button and authenticated)

Reply:
For example, you can try this:
```javascript
  watchEffect(async () => {
    if (this.$auth.isAuthenticated.value) {
      const idTokenClaims = await this.$auth.getIdTokenClaims()
      this.username = response.nickname
      const response = await this.$http.get(`/students/${this.username}`)
      this.student = response
    }
  });


```

(I haven't ran or tested this code, there are probably better ways